core/ui/src/main/java/com/tangem/core/ui/components/bottomsheets/TangemBottomSheet.kt

```diff
@@ -3,29 +3,52 @@ package com.tangem.core.ui.components.bottomsheets
 import androidx.compose.foundation.layout.ColumnScope
 import androidx.compose.material3.ExperimentalMaterial3Api
 import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.SheetState
 import androidx.compose.material3.rememberModalBottomSheetState
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
 import com.tangem.core.ui.res.TangemTheme
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
 
 /**
  * Tangem bottom sheet with custom draggable header and config
  *
- * @param config data model containing logic and ui models
+ * @param config  data model containing logic and ui models
  * @param content custom bottom sheet content
  */
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun TangemBottomSheet(
+inline fun <reified T : TangemBottomSheetConfigContent> TangemBottomSheet(
     config: TangemBottomSheetConfig,
-    content: @Composable ColumnScope.(TangemBottomSheetConfigContent) -> Unit,
+    crossinline content: @Composable ColumnScope.(T) -> Unit,
 ) {
-    ModalBottomSheet(
-        onDismissRequest = config.onDismissRequest,
-        sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true),
-        containerColor = TangemTheme.colors.background.primary,
-        shape = TangemTheme.shapes.bottomSheetLarge,
-        dragHandle = { TangemBottomSheetDraggableHeader() },
-    ) {
-        content(config.content)
+    var isVisible by remember(key1 = true) { mutableStateOf(value = config.isShow) }
+
+    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
+    if (isVisible && config.content is T) {
+        ModalBottomSheet(
+            onDismissRequest = config.onDismissRequest,
+            sheetState = sheetState,
+            containerColor = TangemTheme.colors.background.primary,
+            shape = TangemTheme.shapes.bottomSheetLarge,
+            dragHandle = { TangemBottomSheetDraggableHeader() },
+        ) {
+            content(config.content)
+        }
+    }
+
+    LaunchedEffect(key1 = config.isShow) {
+        if (config.isShow) {
+            isVisible = true
+        } else {
+            sheetState.collapse { isVisible = false }
+        }
+    }
```

Comment:
Кажется, что можно не добавлять `isVisible` и просто проверять поле `config.isShow`
```suggestion

    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
    if (config.isShow && config.content is T) {
        ModalBottomSheet(
            onDismissRequest = config.onDismissRequest,
            sheetState = sheetState,
            containerColor = TangemTheme.colors.background.primary,
            shape = TangemTheme.shapes.bottomSheetLarge,
            dragHandle = { TangemBottomSheetDraggableHeader() },
        ) {
            content(config.content)
        }
    }
```

Reply:
Эта таска была направлена на то, чтобы добавить анимацию скрытия боттом щита (в названии ПР описал).
В твоем варианте ты как раз таки убрал одно из главных изменений.

Я исправлял по гайдлайну материала, так что здесь все верно :)

Reply:
Понял
Тогда в `remember` ключ не нужен, там статичный `true` сейчас